### PR TITLE
add icon selected state

### PR DESCRIPTION
**Pull request by idealllee**
1. allow animation when zooming
2. use MarkerImage to set halo

@jlo1 @drhops

### CODE REVIEW

File: android/src/main/java/com/github/wuxudong/rncharts/charts/ChartBaseManager.java [hunk 1]
```diff
@@ -308,6 +314,83 @@ public void setMarker(Chart chart, ReadableMap propMap) {
...
+            new IMarkerConditionFn() {
+
+                @Override
+                public boolean call(Entry e, Highlight h) {
+                    boolean skip = true;
+                    Map entryData = (e.getData() instanceof Map) ? (Map)e.getData() : new HashMap();
+                    List goalIds = entryData.containsKey("goalIds") ? ((List)entryData.get("goalIds")) : null;
+                    Object highlightMeta = (h instanceof HighlightWithMeta) ? ((HighlightWithMeta) h).getMetaData() : null;
+
+                    if (goalIds != null) {
```
Comment (jlo1) [inline android/src/main/java/com/github/wuxudong/rncharts/charts/ChartBaseManager.java]: wonder if there's a way to support this without needing to rely on switchback logic (e.g. being agnostic to what `goalIds` means, and what `skipGroupedGoals` means?

e.g. maybe within each DataEntry, we can add `{:highlightMeta {:autoMark false}}` for the grouped goals, and `{:highlightMeta {:highlightEnabled false}}` for the non-goal data entries.

Comment (idealllee) [reply android/src/main/java/com/github/wuxudong/rncharts/charts/ChartBaseManager.java]: great suggestion. updated!

File: android/src/main/java/com/github/wuxudong/rncharts/markers/RNConditionalMarkerImage.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    private Map<Integer, Set<Integer>> mExcludes;
+    
+    private Map<Integer, Set<Integer>> parseExcludesProp(ReadableArray excludesProp) {
+        if (excludesProp != null) {
+            List excludeList = ConversionUtil.toList(excludesProp);
+            Map<Integer, Set<Integer>> excludes =  new HashMap<Integer, Set<Integer>>();
+
+            for (Object exclude : excludeList) {
+                Map excludeMap = (Map)exclude;
+                List dataSetIndexesArray = (List)excludeMap.get("dataSetIndexes");
```
Comment (jlo1) [inline android/src/main/java/com/github/wuxudong/rncharts/markers/RNConditionalMarkerImage.java:53]: how does `excludeMap` work with both `dataSetIndexes` and `dataIndex`? If my data looks ...
```
{:dataIndex0 {:dataSets [A B]}
 :dataIndex1 {:dataSets [C D]}}
```
and want to exclude A and D, then it seems that the current implementation wouldn't support it?

File: android/src/main/java/com/github/wuxudong/rncharts/markers/RNConditionalMarkerImage.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+        } else {
+
+            Map hMeta = (h instanceof HighlightWithMeta) ? ((HighlightWithMeta) h).getMetaData() : null;
+            String hSrc = (hMeta != null && hMeta.containsKey("source")) ? (String) hMeta.get("source") : "tap";
+
+            Map eData = (e.getData() instanceof Map) ? (Map)e.getData() : null;
+            String eMarkerSrc = (String) eData.get("markerHighlightSource");
+
+            if (eMarkerSrc != null // default don't skip
```
Comment (jlo1) [inline android/src/main/java/com/github/wuxudong/rncharts/markers/RNConditionalMarkerImage.java:105]: feels much cleaner! 🙌

File: android/src/main/java/com/github/wuxudong/rncharts/charts/BarLineChartBaseManager.java [hunk 4]
```diff
@@ -125,13 +128,32 @@ public void setZoom(BarLineChartBase chart, ReadableMap propMap) {
...
                     (float) propMap.getDouble("scaleX"),
                     (float) propMap.getDouble("scaleY"),
                     (float) propMap.getDouble("xValue"),
                     (float) propMap.getDouble("yValue"),
                     axisDependency
-            );
+                );
+            } else {
+                // TODO(chenyu): use zoomAndCenterAnimated when bug fixed or branch out:
+                // https://github.com/PhilJay/MPAndroidChart/pull/2856
```
Comment (jlo1) [inline android/src/main/java/com/github/wuxudong/rncharts/charts/BarLineChartBaseManager.java:146]: thanks for linking!

**Comment by jlo1** [comment]
LGTM after question about how `excludes` is passed in! 🦀

File: android/src/main/java/com/github/wuxudong/rncharts/markers/RNConditionalMarkerImage.java (hunk 2 above)
Comment (idealllee) [reply android/src/main/java/com/github/wuxudong/rncharts/markers/RNConditionalMarkerImage.java:53]: you would pass in a excludes that looks like:
```
[{:dataIndex 0 :dataSetIndexes [A]}
 {:dataIndex 1 :dataSetIndexes [D]}]
```
Would that work?

Comment (jlo1) [reply android/src/main/java/com/github/wuxudong/rncharts/markers/RNConditionalMarkerImage.java:53]: ahh, yes, sorry! Misread, and didn't realize we were passing through the array of maps! thanks!

**Pull request merged**